tests/test_postfix.py

```diff
@@ -0,0 +1,387 @@
+                    "SMTP_PORT": "1025",
+                    "VIRTUAL_MBOX": "1",
+                    "VMAIL_UID": "5000",
+                    "VIRTUAL_DOMAINS": "example.com example1.com",
+                    "VIRTUAL_USERS": "user1@example.com user2@example.com user@example1.com",
+                },
+                image_format=ImageFormat.DOCKER,
+            ),
+            marks=marks,
+        )
+    )
+
+
+for postfix_ctr in POSTFIX_WITH_VIRTUAL_MBOX_ENABLED:
+    ctr, marks = container_and_marks_from_pytest_param(postfix_ctr)
+    POSTFIX_WITH_LDAP_ENABLED.append(
+        pytest.param(
+            DerivedContainer(
+                base=ctr,
+                forwarded_ports=ctr.forwarded_ports,
+                containerfile=CONTAINERFILE_POSTFIX_WITH_LDAP_ENABLED,
+                extra_environment_variables={
+                    "SERVER_HOSTNAME": "localhost",
+                    "SMTP_RELAYHOST": "localhost",  # pointing to mailhog running locally
+                    "SMTP_PORT": "1025",
+                    "VIRTUAL_MBOX": "1",
+                    "VMAIL_UID": "5000",
+                    "USE_LDAP": "1",
+                    "LDAP_BASE_DN": "dc=example,dc=com",
+                    "LDAP_BIND_DN": "cn=admin,dc=example,dc=com",
+                    "LDAP_BIND_PASSWORD": "admin",
+                    "LDAP_USE_TLS": "0",
+                    "LDAP_TLS": "0",
+                    "LDAP_DOMAIN": "example.com",
+                    "LDAP_ADMIN_PASSWORD": "admin",
+                    "LDAP_CONFIG_PASSWORD": "config",
+                    "SETUP_FOR_MAILSERVER": "1",
+                    "MAIL_ACCOUNT_READER_PASSWORD": "admin",
+                    "LDAP_SERVER_URL": "ldap://localhost",
+                },
+                image_format=ImageFormat.DOCKER,
+            ),
+            marks=marks,
+        )
+    )
+
+
+def test_postfix_status(auto_container):
+    """check if Postfix service is running inside the container"""
+
+    # verify PID 1 process is ENTRYPOINT - `/bin/bash /entrypoint/entrypoint.sh postfix start`
+    assert (
+        "/bin/bash /entrypoint/entrypoint.sh postfix start"
+        in auto_container.connection.check_output(
+            "ps -eo pid,cmd | grep '^ *1 ' | sed 's/^ *1 //'"
+        )
+    )
+
+    assert (
+        "the Postfix mail system is running"
+        in auto_container.connection.check_output("postfix status 2>&1")
+    )
+
+
+@pytest.mark.parametrize(
+    "container_per_test", POSTFIX_WITH_LOGGING_ENABLED, indirect=True
+)
+def test_postfix_send_email_delivered(
+    container_per_test: ContainerData,
+) -> None:
+    """test sending an email successfully to localhost, using Postfix container"""
+
+    assert container_per_test.connection.file("/var/log/postfix.log").exists
+
+    sendmail_cmd = 'echo "Subject: Test Email\n\nThis is a test email body." | sendmail -v root@localhost'
+    assert container_per_test.connection.run_expect([0], sendmail_cmd)
```

Comment:
No need to assert the return value, both `run_expect([0], $cmd)` & `check_output($cmd)` fail when the return value is nonzero

```suggestion
    container_per_test.connection.check_output(sendmail_cmd)
```